jobs/src/main/scala/dcos/metronome/scheduler/impl/NotifyOfTaskStateOperationStep.scala

```diff
@@ -1,56 +1,30 @@
 package dcos.metronome
 package scheduler.impl
 
-import java.time.Clock
-
+import akka.Done
 import akka.event.EventStream
 import dcos.metronome.eventbus.TaskStateChangedEvent
 import dcos.metronome.scheduler.TaskState
-import mesosphere.marathon.core.task.bus.TaskChangeObservables.TaskChanged
-import mesosphere.marathon.core.task.update.TaskUpdateStep
-import mesosphere.marathon.core.task.{ TaskStateChange, TaskStateOp }
+import mesosphere.marathon.core.instance.update.{ InstanceChange, InstanceChangeHandler }
+import java.time.Clock
 
 import scala.concurrent.Future
 
-class NotifyOfTaskStateOperationStep(eventBus: EventStream, clock: Clock) extends TaskUpdateStep {
+class NotifyOfTaskStateOperationStep(eventBus: EventStream, clock: Clock) extends InstanceChangeHandler {
   override def name: String = "NotifyOfTaskStateOperationStep"
 
-  override def processUpdate(taskChanged: TaskChanged): Future[_] = {
-    taskState(taskChanged).foreach { state =>
+  override def process(instanceChange: InstanceChange): Future[Done] = {
+    taskState(instanceChange).foreach { state =>
       val event = TaskStateChangedEvent(
-        taskId = taskChanged.taskId,
+        taskId = instanceChange.instance.appTask.taskId,
         taskState = state,
         timestamp = clock.instant())
       eventBus.publish(event)
     }
 
-    Future.successful(())
+    Future.successful(Done)
   }
 
-  // FIXME: we shouldn't need a translation from TaskChanged to a TaskState - Marathon should provide clear information
-  // about each tasks state without the need to map over options or match case classes (mesosStatus, launched or Status)
-  // for now, this is a super ugly translation:
-  private[this] def taskState(taskChanged: TaskChanged): Option[TaskState] = {
-    (taskChanged.stateOp, taskChanged.stateChange) match {
-
-      // A Mesos status update disregarding the effect
-      case (TaskStateOp.MesosUpdate(_, _, mesosStatus, _), _) => Some(TaskState(mesosStatus))
-
-      // a new launched task
-      case (TaskStateOp.LaunchEphemeral(_), TaskStateChange.Update(_, _)) => Some(TaskState.Created)
-
-      // a new launched task
-      case (_: TaskStateOp.LaunchOnReservation, TaskStateChange.Update(_, _)) => Some(TaskState.Created)
-
-      // Whatever lead to a TaskStateChange.Update if we have a mesosStatus
-      case (_, TaskStateChange.Update(task, _)) if task.mesosStatus.isDefined => Some(TaskState(task.mesosStatus.get))
-
-      // Whatever lead to a TaskStateChange.Update if we have no mesosStatus
-      case (_, TaskStateChange.Update(_, _)) => Some(TaskState.Created)
-
-      // Whatever
-      case _ => None
-    }
-  }
+  private[this] def taskState(instanceChange: InstanceChange): Option[TaskState] = TaskState(instanceChange.condition)
```

Comment:
I actually decided not to do that because then I could not do this "c if c.isLost" - that way I don't need to know of all the conditions and can work with the lost conditions as a group